File: 1403. Minimum Subsequence in Non-Increasing Order.py
class Solution(object):
    def minSubsequence(self, nums):
        res = []
        nums.sort(reverse=True)
        summ = sum(nums)
        cur = 0
        for i in range(len(nums)):
            cur += nums[i]
            res.append(nums[i])
            if cur > summ - cur:
                return res
        

File: 1403. Minimum Subsequence in Non-Increasing Order.ts
function minSubsequence(nums: number[]): number[] {
    const res:number[] = [];
    nums.sort((a, b) => b - a);
    let sum:number = nums.reduce((a,b)=>a+b), cur:number = 0;
    for (let i:number = 0; i < nums.length; i++) {
        cur += nums[i];
        res.push(nums[i]);
        if (cur > sum - cur) return res;
    }
};
